lib/models/application-invite.ts

```diff
@@ -0,0 +1,280 @@
+You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import * as errors from 'balena-errors';
+import * as Promise from 'bluebird';
+import {
+	Application,
+	ApplicationInvite,
+	ApplicationMembershipRole,
+	PineOptionsFor,
+} from '../../typings/balena-sdk';
+import { InjectedDependenciesParam, InjectedOptionsParam } from '../balena';
+import { findCallback, mergePineOptions } from '../util';
+
+const RESOURCE = 'invitee__is_invited_to__application';
+
+const getApplicationInviteModel = function(
+	deps: InjectedDependenciesParam,
+	opts: InjectedOptionsParam,
+) {
+	const { request, pine } = deps;
+	const { apiUrl } = opts;
+
+	exports = {};
+
+	/**
+	 * @summary Get a single application invite
+	 * @name get
+	 * @public
+	 * @function
+	 * @memberof balena.models.applicationInvite
+	 *
+	 * @param {Number} id - applicationInvite id
+	 * @param {Object} [options={}] - applicationInvite pine options
+	 * @fulfil {Object} - applicationInvite
+	 * @returns {Promise}
+	 *
+	 * @example
+	 * balena.models.applicationInvite.get(51).then(function(applicationInvite) {
+	 * 	console.log(applicationInvite);
+	 * });
+	 *
+	 * @example
+	 * balena.models.applicationInvite.get(51, function(error, applicationInvite) {
+	 * 	if (error) throw error;
+	 * 	console.log(applicationInvite);
+	 * });
+	 */
+	exports.get = function(
+		id: number,
+		options: PineOptionsFor<ApplicationInvite> = {},
+		callback?: (error?: Error, result?: ApplicationInvite) => void,
+	): Promise<ApplicationInvite> {
+		callback = findCallback(arguments);
+		return pine
+			.get({
+				resource: RESOURCE,
+				id,
+				options: mergePineOptions(
+					{
+						$select: [
+							'id',
+							'is_invited_to__application',
+							'invitee',
+							'message',
+							'created_at',
+						],
+					},
+					options,
+				),
+			})
+			.asCallback(callback);
+	};
+
+	/**
+	 * @summary Get all Invites by application
+	 * @name getAll
+	 * @public
+	 * @function
+	 * @memberof balena.models.applicationInvite
+	 *
+	 * @description
+	 * This method returns all invites of a specific application.
+	 *
+	 * @param {number} applicationId - Application Id for w
+	 * @param {Object} [options={}] - extra pine options to use
+	 * @fulfil {Object[]} - applicationInvites
+	 * @returns {Promise}
+	 *
+	 * @example
+	 * balena.models.applicationInvite.getAllByApplication(applicationId).then(function(applicationInvites) {
+	 * 	console.log(applicationInvites);
+	 * });
+	 *
+	 * @example
+	 * balena.models.applicationInvite.getAllByApplication(applicationId, function(error, applicationInvites) {
```

Comment:
Again, I'll leave it to Thodoris to say what he prefers, but I'd say we don't have to add callback examples.